server.js

```diff
@@ -193,11 +189,132 @@ app.get('/dashboards', function(req, res) {
 app.get('/bugs', function(req, res) {
   res.sendFile(path.join(__dirname, './app/public/index.html'));
 });
+app.get('/issues', function(req, res) {
+  res.sendFile(path.join(__dirname, './app/public/index.html'));
+});
 
+// Cache check middleware: if the URL is in cache, use that.
+function checkCache( req, res, next ) {
+  if ( checkCache.overrides[ req.url ] ) {
+    delete checkCache.overrides[ req.url ];
+    next();
+    return;
+  }
+  cache.read( req.url, function( err, data ) {
+    if ( err || !data ) {
+      next( err );
+      return;
+    }
+    res.json( data );
+  });
+}
+checkCache.overrides = {};
+
+var mozillaRepos = "id.webmaker.org webmaker-curriculum snippets teach.webmaker.org goggles.webmaker.org webmaker-tests sawmill login.webmaker.org openbadges-badgekit webmaker-app api.webmaker.org popcorn.webmaker.org webmaker-mediasync webmaker.org webmaker-app-cordova webmaker-metrics nimble mozilla-opennews teach-api mozillafestival.org call-congress-net-neutrality thimble.webmaker.org advocacy.mozilla.org privacybadges webmaker-profile-2 call-congress build.webmaker.org webmaker-landing-pages webliteracymap events.webmaker.org badgekit-api openbadges-specification make-valet webmaker-auth webmaker-events-service webmaker-language-picker MakeAPI blog.webmaker.org webmaker-login-ux webmaker-desktop webmaker-app-publisher badges.mozilla.org lumberyard webmaker-download-locales webmaker-addons bsd-forms-and-wrappers popcorn-js hivelearningnetworks.org webmaker-firehose makeapi-client makerstrap webmaker-app-bot webmaker-screenshot react-i18n webmaker-kits-builder webmaker-app-guide".split(" ");
+var orgs = ["MozillaFoundation", "MozillaScience"];
+
+app.get( "/api/github/mozilla-repo-names", checkCache, function(req, res) {
+  // Get Foundation repos then merge them with a static list of mozilla repos.
+  github.getRepos(orgs, function(err, results) {
+    if (err) {
+      // Not sure what to do with errors, yet.
+      console.log(err);
+    } else {
+      var repoNames = [];
+      results.forEach(function(repo) {
+        repoNames.push(repo.full_name);
+      });
+      // Merge with static list of mozilla repos.
+      res.json( repoNames.concat(mozillaRepos.map(function(item) {
+        return "mozilla/" + item;
+      })));
+    }
+  });
+});
+app.get( "/api/github/foundation-users", checkCache, function(req, res) {
+  github.getUsersForOrgs(orgs, function(err, results) {
+    if (err) {
+      // Not sure what to do with errors, yet.
+      console.log(err);
+    } else {
+      res.json(results);
+    }
+  });
+});
+app.get( "/api/github/mozilla-labels", checkCache, function(req, res) {
+  var url = "http://127.0.0.1:" + app.get('port') + "/api/github/mozilla-repo-names";
```

Comment:
I found a better way, it wasn't sitting well with me either.

I *was* doing it this way because the cache I was using was directly linked to the http call.

I ended up building the cache into API instead. I initially didn't want too much cache code mixed in with github code but I reduced it down to one spot which sits with me nicely. https://github.com/ScottDowne/build.webmaker.org/blob/a530713baed53700a9629083831d4427f86040a0/server/models/github.js#L95-L98